packages/notifications/push/amplify_push_notifications/lib/src/amplify_push_notifications_impl.dart

```diff
@@ -281,11 +341,51 @@ class _PushNotificationsFlutterApi implements PushNotificationsFlutterApi {
     unawaited(_flushEvents());
   }
 
+  void setProvider(ServiceProviderClient serviceProviderClient) {
+    // Flush only when the SericeProvider becomes available in config
+    _serviceProviderClient = serviceProviderClient;
+    unawaited(_flushEvents());
```

Comment:
Hum this makes emitting PN in the background depends on the pinpoint client... if for some reason pinpoint client cannot be set, the PN will stop working completely... Is that what we want?

Reply:
Changed this to only have flushRecordEvents. And always emit to callbacks when notification is received.